components/org.wso2.carbon.identity.application.authenticator.oauth2/src/main/java/org.wso2.carbon.identity.application.authenticator.oauth2/Oauth2GenericAuthenticator.java

```diff
@@ -487,10 +486,9 @@ public String getContextIdentifier(HttpServletRequest request) {
         }
     }
 
-    protected String generateState() {
```

Comment:
Deprecate this and overload this function into the new one with param context and keep both.

Reply:
Addressed from https://github.com/wso2-extensions/identity-outbound-auth-oauth2/pull/20